src/blocks/weather.rs

```diff
@@ -126,6 +126,10 @@ impl Weather {
 
                 let raw_temp = json.pointer("/main/temp").and_then(|v| v.as_f64()).ok_or_else(malformed_json_error)?;
 
+                let raw_humidity = json.pointer("/main/humidity")
+                    .and_then(|v| v.as_f64())
```

Comment:
It's likely that this will be missing in some locations. Best to use `.map_or(Some(0.0), |v| v.as_f64())` instead, as for wind speed below.